### PR TITLE
chore(docs): Update the link to global css example

### PR DESCRIPTION
**Pull request by pvaladez**
## Description

On line 13, the link to index.js points to a commit from 2018.  The link is there to demonstrate how a Layout component is implemented, and this implementation has changed from 2018 to 2020.  For me, the way it's done in the latest commit is much easier to understand.

Also, when I clicked on the link from documentation, at first I didn't realize that I wasn't looking at the latest commit and it caused a bit of confusion for me.

Not sure if you want to link to the [commit on May 28th, 2020:](https://github.com/gatsbyjs/gatsby-starter-default/blob/e1e2a960f8a54e2a1234d1694d390b1cc26b7947/src/pages/index.js#L9) 

Or just link to the [latest commit:](https://github.com/gatsbyjs/gatsby-starter-default/blob/master/src/pages/index.js#L9) 

In the proposed change, I updated the link to the May 28th commit to be safe.

### Documentation

[https://www.gatsbyjs.com/docs/global-css/](https://www.gatsbyjs.com/docs/global-css/)

### CODE REVIEW

**Comment by acagastya** [comment]
This PR just does not make any sense.  The file is talking about CSS, and your PR now links to a React Component.

@pvdz @alanmatw can you please close this PR and flag it as invalid?

**Comment by pvaladez** [comment]
The first subheading(h2) on the page is about how to use a Layout [React] component to implement global CSS.  See: https://www.gatsbyjs.com/docs/global-css/#adding-global-styles-with-a-layout-component

The original link pointed to a React component... I am only updating the link's commit date from Apr 9, 2018 to May 28, 2020.  Furthermore, it might make sense to point the link to the latest commit, i.e. [master](https://github.com/gatsbyjs/gatsby-starter-default/blob/master/src/pages/index.js#L9), instead of linking to a particular commit.

**Comment by acagastya** [comment]
The global CSS was located at [line 6](https://github.com/gatsbyjs/gatsby-starter-default/blob/02324e5b04ea0a66d91c7fe7408b46d0a7eac868/src/layouts/index.js#L6).  The correct line is `import './index.css'`, not the React Component.

Moreover, linking to master is NOT a good idea -- a newer revision in master may change the line number, and in the newer revisions, `import './index.css'` may not be on the same line.

**Comment by pvaladez** [comment]
Ok... I understand why you would be confused by that, but please hear me out...  

The heading for this section of the documentation is **"Adding global styles with a layout component"** and the paragraph under it is likewise describing the use of a Layout component for the **purpose** of loading css globally... the focus of the paragraph and the Note that follows is centered on using a **Layout component**... I don't think the focus is on the use of "import" in order to load the css.  

Therefore, I put the link's focused line(Line 9) on the Layout component.  I would understand if you want to make the argument that the link should point to the [use of "import" in the layout.js file](https://github.com/gatsbyjs/gatsby-starter-default/blob/5549a3d8f5d1733fed766a998154a1693e42656c/src/components/layout.js#L13), but I left the link pointing to index.js because that is where the last link was pointed... Also, I think it is easier to first see where the <Layout> component is used and find out how it is implement afterward rather than the reverse... if you start them off in the layout.js file then they may not immediately know where it's used... on the other hand they can easily see where the Layout component is implemented by looking at [Line 4](https://github.com/gatsbyjs/gatsby-starter-default/blob/master/src/pages/index.js#L4) of index.js

Regardless on your opinion about the line number or the file that the link should point to...  the link should point to how  [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default) implements a **"shared layout with global styles"** in 2020, not 2018.

Lastly, should the link always point to the latest commit of [gatsby-starter-default](https://github.com/gatsbyjs/gatsby-starter-default) rather than linking to a particular commit date?

**Comment by pvaladez** [comment]
One more point: most people who are trying to learn how Gatsby works aren't going to click on the [link](https://github.com/gatsbyjs/gatsby-starter-default/blob/02324e5b04ea0a66d91c7fe7408b46d0a7eac868/src/layouts/index.js#L6), and only read the one line that is highlighted... they are going to read the whole file.  And if the solution does only require one line of code then that line of code should be put directly into the documentation instead of linking to it.  

In this case, the one highlighted line of code:

`import './index.css'`

does not at all demonstrate how the "pattern" of using a "shared layout component" "to add global styles" is done in [the default starter](https://github.com/gatsbyjs/gatsby-starter-default/blob/02324e5b04ea0a66d91c7fe7408b46d0a7eac868/src/layouts/index.js#L6)

**Comment by pvaladez** [comment]
One more thing... the need to point the link to a 2020 commit isn't just because the code is two years newer... 

The way that the layout component was implemented in the 2018 commit looks like it no longer works... correct me if I'm wrong but it looks like this implementation was a **Gatsby v1 special "layout" component**.  This "special layout component" was removed in Gatsby v2: [See here](https://github.com/gatsbyjs/rfcs/blob/master/text/0002-remove-special-layout-components.md)

Besides that... I still don't understand how these special layout components in v1 work, and I don't understand why the file for the Layout component would be named index.js instead of layout.js.  

There is a [better way](https://github.com/gatsbyjs/gatsby-starter-default/blob/e1e2a960f8a54e2a1234d1694d390b1cc26b7947/src/pages/index.js#L9) in 2020 with Gatsby v2.

**Comment by pvaladez** [comment]
More explanation and confirmation of my last comment: https://www.gatsbyjs.com/blog/2018-06-08-life-after-layouts/

File: docs/docs/global-css.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Globally-scoped CSS rules are declared in external `.css` stylesheets, and [CSS
 
 The best way to add global styles is with a [shared layout component](/tutorial/part-three/#your-first-layout-component). This layout component is used for things that are shared throughout the site, including styles, header components, and other common items.
 
-> **NOTE:** This pattern is implemented by default in [the default starter](https://github.com/gatsbyjs/gatsby-starter-default/blob/02324e5b04ea0a66d91c7fe7408b46d0a7eac868/src/layouts/index.js#L6).
+> **NOTE:** This pattern is implemented by default in [the default starter](https://github.com/gatsbyjs/gatsby-starter-default/blob/e1e2a960f8a54e2a1234d1694d390b1cc26b7947/src/pages/index.js#L9).
```
Comment (LekoArts) [inline docs/docs/global-css.md:13]:
```suggestion
> **NOTE:** This pattern is implemented by default in [the default starter](https://github.com/gatsbyjs/gatsby-starter-default/blob/063978d59f74103da45d5880a61ebd2e77798e3c/src/components/layout.js#L13).
```

**Review by LekoArts — CHANGES_REQUESTED** [review]
Thanks for the PR! I still think that we should link to the `'./layout.css'` section instead of the React component. Updating the link itself makes sense as you correctly pointed out the pattern for layouts has changed since then.

**Review by LekoArts — APPROVED** [review]
Thanks! 👍

**Comment by pvaladez** [comment]
Thank you for taking the time to understand what I was asking for!  Gatsby's documentation on the whole is impressively understandable, comprehensive and has been super helpful to me, so I'm excited that I could contribute.

Sorry to @acagastya for the confusion, and actually @acagastya's response made me realize that I had made some incorrect assumptions about exactly **what** the documentation was trying to convey and exactly **how** Gatsby loads CSS.  Afterward, I realized that this doc page might benefit from further changes.

Specifically, I think the documentation is missing information on how Gatsby loads CSS on a production site, and how that affects what people may regard as "Global CSS".

Below is just my thoughts on what might help the documentation be more clear... I think any potential changes belong in a separate PR?  But I am writing here because it seems related and maybe I could get feedback if you think such changes would be appropriate... I'm also happy to move this to a new issue or PR.

### Background:  
On a traditional static/html/css website, every page you visit needs to load all resources separately for every page request.  So if you want a particular CSS file to be used "globally", i.e. on all html pages, you need to load the global CSS file either via a link element or an inline style tag on every page.  

However, the default behavior in Gatsby is such that it combines ALL the CSS used everywhere, minifies it, and places that CSS in an inline style tag in the head.  This means that when you import CSS via `import 'globalCSSFile.css;'` in ANY component in Gatsby, that CSS is now available everywhere/globally.  Conceptually, it makes sense to define these imports in a shared layout component, but technically it's not required for the CSS to be included on all "pages".

### My Confusion:
When I was developing a Gatsby site locally using `gatsby develop`, I noticed in Chrome Dev Tools that there were `<link href="blob:">` tags being inserted in the head for each "import" of CSS files in my various React Components.  My incorrect assumption was that Gatsby dynamically loads CSS in the same way on a production build.  Therefore, when I read the [documentation](https://www.gatsbyjs.com/docs/global-css/), I thought it was saying that the shared layout component was directly enabling the loading of "Global CSS Files"... I now see that this isn't exactly correct.

### Potential Clarification:
Create a note or a paragraph at the top of this doc page to explain how a production build of Gatsby loads CSS with its default behavior.  Because of the way that Gatsby combines all CSS and inserts it into an inline style tag, ALL CSS is therefore available on all pages.  Placing the CSS in a shared Layout component is a good practice, because it allows you to ensure the correct order or CSS rules via the order of the "imports", i.e. you can make sure a "reset.css" file is loaded before any other CSS.  A shared Layout component also makes it easy to include a header and footer on all pages.

The [documentation](https://www.gatsbyjs.com/docs/global-css/) currently discusses methodologies on how to manage the Global Nature of CSS **once it is loaded into the browser**... CSS Modules fixes this problem whereas the BEM naming convention solves it for regular CSS... but I haven't read anything in Gatsby documentation that explains **how and when CSS is loaded**.  If you are new to Gatsby, you might forget and be confused by some ways that Gatsby is different from plain html sites, so I think would be helpful to contrast traditional website CSS file loading with the way CSS is loaded on a Gatsby site running with javascript where the page is essentially an SPA and there is no opportunity to "unload" CSS files out of the browser.  Even when runing Gatsby without javascript, the the combined/minified CSS is in the inline style tag of every page... so the takeaway is that... by default... All CSS is loaded everywhere.

### Feedback?
Do you see anything wrong with my reasoning, and should I open a new PR adding the information I suggested?

**Pull request merged**